### PR TITLE
Fix uploding JSON data for external cluster connection strict checking issue

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
For more info: https://bugzilla.redhat.com/show_bug.cgi?id=2092217

Signed-off-by: Gowtham Shanmugasundaram <gshanmug@redhat.com>

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
image

File: packages/odf/components/utils/common.ts [hunk 1]
```diff
@@ -211,9 +211,11 @@ export const checkError = (
   });
 
   // Check for missing keys
-  const missingKeys = _.difference(
-    _.concat(requiredKeys, requiresEncodingKeys),
-    providedKeys
```
Comment (bipuladh) [inline packages/odf/components/utils/common.ts:216]: Could `_.deepFlatten(providedKeys)` work?

Comment (GowthamShanmugam) [reply packages/odf/components/utils/common.ts:216]: Do you mean _.flattenDeep? 

it is for converting the multidimensional list to a single multidimensional. problem is providedKey keys have some postfix. we can't do direct differences, instead, we need to compare the substring.

**Review by bipuladh — APPROVED** [review]

**Pull request merged**